camel/models/openai_model.py

```diff
@@ -48,10 +48,16 @@ def run(
             Dict[str, Any]: Response in the OpenAI API format.
         """
         import openai
-        messages_openai: List[OpenAIMessage] = messages
-        response = openai.ChatCompletion.create(messages=messages_openai,
-                                                model=self.model_type.value,
-                                                **self.model_config_dict)
+        if self.model_type == ModelType.TEXT_DAVINCI_003:
+            prompt = messages[-1]['content']
```

Comment:
Raise on empty messages and keys not there.

Reply:
Thanks, done : )